### PR TITLE
STR-90--BE: Make middleware on backend router

### PR DESCRIPTION
**Pull request by mbendandour**
# Description
I added code to the Handler.php that would route a user to a 404 JSON if they typed in a route that doesn't exist. Moreover, if they are logged in and type in a route that doesn't exist, they should get routed back to the login page  
# Testing
Remove the # in the url and type in '/helloo' in the route. You should see a 404. Next, log in as steve. Remove the #, and type in any non-existent url, like '/blah'. This should take you back to the login page
if you want to continue testing the 404 error, make sure to add '/logout' to the route to log out first
# Installation
no

### CODE REVIEW

**Review by liadGolan — CHANGES_REQUESTED** [review]
Middleware should not log a person a out if they go outside the view app, if they are still detected as logged in, they should be directed to the application.

If someone is not logged in, they should be redirected to the front end page rather than being given a JSON with 404

**Review by liadGolan — APPROVED** [review]
👍

File: app/Http/Kernel.php [hunk 1]
```diff
@@ -30,7 +31,7 @@ class Kernel extends HttpKernel
         'web' => [
             \App\Http\Middleware\EncryptCookies::class,
             \Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse::class,
-            \Illuminate\Session\Middleware\StartSession::class,
+//            \Illuminate\Session\Middleware\StartSession::class,
```
Comment (luisjg) [inline app/Http/Kernel.php]: We should remove it from this list

**Review by luisjg — CHANGES_REQUESTED** [review]
See comments

**Review by luisjg — APPROVED** [review]
👍

**Pull request merged**